File: app/src/main/java/com/tech/dagger2_android/inJava/basic/MainViewModel.java
package com.tech.dagger2_android.inJava.basic;

import com.tech.dagger2_android.inJava.network.NetworkClient;

import javax.inject.Inject;

public class MainViewModel {

    private NetworkClient mClient ;

    @Inject
    public MainViewModel(NetworkClient networkClient){
       mClient = networkClient;
    }
    public String fetchData(){
        return mClient.fetchData();
    }
}

File: app/src/main/java/com/tech/dagger2_android/inKotlin/basic/MainViewModel.kt
package com.tech.dagger2_android.inKotlin.basic

import com.tech.dagger2_android.inKotlin.network.NetworkClient
import javax.inject.Inject

class MainViewModel @Inject constructor(networkClient: NetworkClient?)  {
    private var mClient: NetworkClient? = null

    init {
        mClient = networkClient
    }

    fun fetchData(): String? {
        return mClient?.fetchData()
    }
}